group-coordinator/src/main/java/org/apache/kafka/coordinator/group/classic/ClassicGroup.java

```diff
@@ -1300,6 +1341,68 @@ public Map<String, byte[]> groupAssignment() {
         ));
     }
 
+    /**
+     * Convert the current classic group to a consumer group.
+     * Add the records for the conversion.
+     *
+     * @param consumerGroup     The converted consumer group.
+     * @param records           The list to which the new records are added.
+     *
+     * @throws GroupIdNotFoundException if any of the group's member doesn't support the consumer protocol.
+     */
+    public void convertToConsumerGroup(
+        ConsumerGroup consumerGroup,
+        List<Record> records,
+        TopicsImage topicsImage
+    ) throws GroupIdNotFoundException {
```

Comment:
The order of the records matter here. We should have the following order:
1. Member subscriptions.
2. Group epoch.
3. Member target assignment.
4. Target assignment epoch.
5. Member current assignments.

Reply:
`Group epoch` and `Target assignment epoch` only apply to the group. Should we add the 5 records for each member?